### PR TITLE
#332 Use ProcessRunner instead of ProcessBuilder

### LINKED ISSUE #332: Refactor ProcessBuilder to ProcessRunner
Follow up of #331

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #332

### CODE REVIEW

File: src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java [hunk 1]
```diff
@@ -376,8 +377,7 @@ public IosSigningConfiguration getIosSigningConfiguration() {
      * @throws IllegalArgumentException when the configuration doesn't contain a property graalPath
      * @throws IOException when the path to bin/native-image doesn't exist
      */
-    public void canRunNativeImage() throws IOException {
-
+    public void canRunNativeImage() throws IOException, InterruptedException {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java]: Codacy Issue found: [The method 'canRunNativeImage()' has an NPath complexity of 224](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430)

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430&bid=16457136)

File: src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java [hunk 2]
```diff
@@ -126,67 +125,60 @@ public boolean compile(String cp) throws IOException, InterruptedException {
...
                 extraMessage = "Objectfile should be called "+nameSearch+" but we didn't find that under "+gvmPath.toString();
             }
         }
         if (failure) {
-            System.err.println("Compilation failed with result = " + result);
-            printFromInputStream(inputStream);
-
-            if (extraMessage!= null) {
-                System.err.println("Additional information: "+extraMessage);
+            if (extraMessage != null) {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430)

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430&bid=16457136)

File: src/main/java/com/gluonhq/substrate/util/ProcessRunner.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ private Process setupProcess(String processName, File directory) throws IOExcept
         return pb.start();
     }
 
-    private static Thread mergeProcessOutput(final InputStream is, final StringBuffer sb) {
+    private static Thread mergeProcessOutput(final InputStream is, final StringBuffer sb, final boolean info) {
```
Comment (tiainen) [inline src/main/java/com/gluonhq/substrate/util/ProcessRunner.java]: What is the use for passing the `sb` and `info` fields to this method? The method is always invoked with `ProcessRunner.answer` and `ProcessRunner.info`, so you might as well use those fields directly?

Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/ProcessRunner.java]: Not sure if I follow? `mergeProcessOutput` is static, but those files are not, so I can't use them?

Comment (tiainen) [reply src/main/java/com/gluonhq/substrate/util/ProcessRunner.java]: Good point, why don't we remove static from the method as well as it is called as if it was an instance method.

Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/ProcessRunner.java]: Ok, makes sense. Pushed

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=4983430&bid=16457136)

**Pull request merged**